examples/radial_gradient_filters.py

```diff
@@ -0,0 +1,57 @@
+"""
+=======================
+Radial Gradient Filters
+=======================
+
+This example applies both the normalizing radial gradient (`sunkit_image.radial.nrgf`) filter and Fourier
+normalizing radial gradient filter (`sunkit_image.radial.fnrgf`) to a sunpy map.
+
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+import astropy.units as u
+
+import sunpy.map
+import sunpy.data.sample
+
+import sunkit_image.radial as radial
+from sunkit_image.utils import equally_spaced_bins
+
+###########################################################################
+
+# Sunpy's sample data contain a number of suitable FITS files for this purpose.
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+
+###########################################################################
+
+# Radial bins are calculated over which the filters will be applied. The distance
+# between 1 Solar radius and 2 Solar radius is divided into 100 equal parts.
+radial_bin_edges = equally_spaced_bins()
+radial_bin_edges *= u.R_sun
+
+# The NRGF filter is applied after it.
+out1 = radial.nrgf(aia_map, radial_bin_edges)
+
+# The NRGF filtered map is plotted. 
+out1.plot()
+plt.show()
+
+###########################################################################
+
+# Assuming values for the parameters of FNRGF
+# Order is the number of Fourier coefficients to be used in the approximation.
+# The attentuation coefficient are calculated to be linearly decreasing, you should
+# choose them according to your requirements.
+order = 20
+attenuation_coefficients = np.zeros((2, order + 1))
+attenuation_coefficients[0, :] = np.linspace(1, 0, order + 1)
+attenuation_coefficients[1, :] = np.linspace(1, 0, order + 1)
```

Comment:
So these always have to be calculated before you use the function?

Reply:
Yes, but I cannot create a fuction for this because here I take the coefficients as always decreasing that may not be the case always. It depends on the user.

Reply:
Is there any rules or guidelines on how these should constructed?

Reply:
No. There are no defined rules or guidelines. Have a look

![Screenshot from 2019-06-10 16-09-47](https://user-images.githubusercontent.com/39871718/59190582-b508fb80-8b9a-11e9-83a8-24fd7ef426b1.png)


![Screenshot from 2019-06-10 16-10-23](https://user-images.githubusercontent.com/39871718/59190547-9efb3b00-8b9a-11e9-8ff5-8f8c00e2f163.png)

Reply:
Ok, we should make sure that is mentioned in the docstring somewhere.

Those two tools it mentions at the end? Could we recreate those?

Reply:
one of them is what I have been doing, I can create a function.
The other is the same thing only the last few terms will be zero as defined by the user.
Both can be done.

Reply:
Create one function and just add a keyword that sets the last few terms to 0?

Reply:
Yes doing the same.

Reply:
That seems to suggest it sets the coefficients up to a certain order and the rest are 0.

Reply:
Okay then what I have written is fine I guess

Reply:
Yeah.